ChristianChiarulli/LunarVim
# [Feature]: Add capability to override keymapping + refactor keymappings

# Description

I'm adding the capability to override the default keymap without replacing the whole of them. The current behaviour is replacing the default keymap entirely (see #1140). Also I'm doing some refactoring on keymappings so that the code is more readable.

The way we can add / override the keymap is just by adding something like this on `lv-config`

```lua
lvim.keys.normal_mode['po'] = '<ESC>' -- use default opt based on mode
lvim.keys.normal_mode['po'] = {'<ESC>', { silent = true, noremap = true } } -- if you want to use custom opt
```

Fixes #1140 

## How Has This Been Tested?

- Add custom keymapping on lvim config:
```lua
lvim.keys.insert_mode["po"] = 'echo "Hello"<CR>'
```
- Check the keymapping is working
- Check default keymapping is working (not replaced)
